fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java

```diff
@@ -183,11 +184,14 @@ public void prepareRelatedMVs(Set<Table> queryTables, Set<MaterializedView> rela
             logMVPrepare(connectContext, "queryExcludingMVNames:{}, queryIncludingMVNames:{}",
                     Strings.nullToEmpty(queryExcludingMVNames), Strings.nullToEmpty(queryIncludingMVNames));
 
-            Set<String> queryExcludingMVNamesSet = Sets.newHashSet(queryExcludingMVNames.split(","));
-            Set<String> queryIncludingMVNamesSet = Sets.newHashSet(queryIncludingMVNames.split(","));
+            final Set<String> queryExcludingMVNamesSet = Strings.isNullOrEmpty(queryExcludingMVNames) ? Sets.newHashSet()
+                    : Arrays.stream(queryExcludingMVNames.split(",")).map(String::trim).collect(Collectors.toSet());
+
+            final Set<String> queryIncludingMVNamesSet = Strings.isNullOrEmpty(queryIncludingMVNames) ? Sets.newHashSet()
+                    : Arrays.stream(queryIncludingMVNames.split(",")).map(String::trim).collect(Collectors.toSet());
```

Comment:
My question is how the spaces are introduced.

Reply:
This is a session variable, it is configured by user, maybe like this
set query_excluding_mv_names = "mv1 , mv2, mv3";
so add trim here to process more cases.